test/bulk2.test.ts

```diff
@@ -1,16 +1,66 @@
 import assert from 'assert';
 import path from 'path';
 import fs from './helper/fs';
-import { Connection, Date as SfDate } from 'jsforce';
+import { Connection, Date as SfDate, Schema, Record } from 'jsforce';
 import ConnectionManager from './helper/connection-manager';
 import config from './config';
 import { isObject, isString } from './util';
 import { isNodeJS } from './helper/env';
+import { BulkOperation, IngestJobV2Results } from 'jsforce/lib/api/bulk';
 
 const connMgr = new ConnectionManager(config);
 const conn = connMgr.createConnection();
 conn.bulk2.pollTimeout = 90000; // set bulk2 poll timeout to 90s for tests in this file
 
+async function insertAccounts(
+  id: string | number,
+  qty: number = 20,
+): Promise<IngestJobV2Results<Schema>> {
+  if (typeof id === 'number') {
+    id = id.toString();
+  }
+
+  const insertRecords = [
+    ...Array.from(Array(qty), () => ({
+      Name: `Bulk Account ${id}`,
+    })),
+  ];
+
+  const res = await conn.bulk2.loadAndWaitForResults({
+    object: 'Account',
+    operation: 'insert',
+    input: insertRecords,
+  });
+
+  ensureSuccessfulBulkResults(res, qty, 'insert');
+
+  return res;
+}
+
+function ensureSuccessfulBulkResults(
+  results: IngestJobV2Results<Schema>,
+  qty: number,
+  operation: BulkOperation,
+) {
+  const { successfulResults, failedResults, unprocessedRecords } = results;
+
+  assert.ok(Array.isArray(successfulResults));
+  assert.ok(successfulResults.length === qty);
+
+  const recordCreated = operation === 'insert' ? 'true' : 'false';
+
+  for (const successfulResult of successfulResults) {
+    assert.ok(isString(successfulResult.sf__Id));
+    assert.ok(successfulResult.sf__Created === recordCreated);
```

Comment:
`sf__Created` will only be `true` if it was a bulk insert, all others operations should return false:
https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/get_job_successful_results.htm?q=sf__Created